src/sentry/identity/vsts/provider.py

```diff
@@ -47,6 +51,21 @@ def get_pipeline_views(self):
             AccountConfigView(),
         ]
 
+    def get_refresh_token_header(self):
+        return {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Content-Length': '1654',
+        }
+
+    def get_refresh_token_params(self, refresh_token):
+        return {
+            'client_assertion_type': 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer',
+            'client_assertion': self.get_oauth_client_secret(),
+            'grant_type': 'refresh_token',
+            'assertion': refresh_token,
+            'redirect_uri': absolute_uri(self.oauth_redirect_url),
```

Comment:
I'm not sure I understand why there's this redirect URL? Doesn't this call happen from our server?

Reply:
So @EvanPurkhiser it's a VSTS thing. If you check out the docs the redirect URL is used as a check. The redirect URL is required to match the redirect url that was registered within the app, so passing it is required. This is true even in the case you're refreshing the token.

More here: https://docs.microsoft.com/en-us/vsts/integrate/get-started/authentication/oauth?view=vsts

Reply:
Let me know if there's a better way of doing it. This was the best I could think to make the minimal changes possible